### PR TITLE
Patch to increase the timestep

### PR DESCRIPTION
**Pull request by rmcabezon**
Changed the way vijsignal is calculated in momentum_energy_kern.hpp
This leads to timestep x1.5-x2 larger. Acceptable changes in conservation for a much reduced total NH for TTS.

### CODE REVIEW

File: sph/include/sph/hydro_ve/momentum_energy_kern.hpp [hunk 1]
```diff
@@ -173,7 +173,7 @@ momentumAndEnergyJLoop(cstone::LocalIndex i, T sincIndex, T K, const cstone::Box
         T viscosity_ij = artificial_viscosity(alpha_i, alpha[j], ci, cj, wij);
 
         // For time-step calculations
-        T vijsignal = ci + cj - T(3) * wij;
+        T vijsignal = T(0.5) * (ci + cj) - T(2) * wij;
```
Comment (jgphpc) [inline sph/include/sph/hydro_ve/momentum_energy_kern.hpp:176]: This change makes the [MomentumEnergy.JLoop](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L172) test to fail:

> EXPECT_NEAR(maxvsignal, 4.5535959472046174e7, 1e-6);

```
The difference between maxvsignal=26490876.319252387 and
4.5535959472046174e7 is 19045083.152793787, which exceeds 1e-6
```

while the previous code passes the test:
```
        T vijsignal = ci + cj - T(3) * wij;
        maxvsignal= 45536000
```
A fix would be to modify [L172](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L172) and [L194](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L194) of the test OR rectify [sph/include/sph/hydro_ve/momentum_energy_kern.hpp](https://github.com/unibas-dmi-hpc/SPH-EXA/pull/387/files/8f891e63f19d67793e6b8701624fa4bef88fc2d9#diff-18c0a64c44c98b69666874dd53ae33110f32a0292bacbc46b85ff8a05b693269)

Comment (jgphpc) [reply sph/include/sph/hydro_ve/momentum_energy_kern.hpp:176]: This change also makes the analytical test to fail:

```
Created solution file: 'sedov_solution_0.116509.dat'
Loaded 125000 particles
Density L1 error 0.3283945983914492 <-------- not OK
Pressure L1 error 0.9280070281034704
Velocity L1 error 0.9429304778530712

* Reason: performance error: failed to meet reference:
Density=0.328394598391449, expected 0.14
(l=0.13790000000000002, u=0.14140000000000003)
```

while the previous code passes:
```
Created solution file: 'sedov_solution_0.068745.dat'
Loaded 125000 particles
Density L1 error 0.14064247357249088 <-------- OK
Pressure L1 error 0.8996386174012747
Velocity L1 error 0.9146858514719047
```

I can share the sedov_solution_0.068745.dat file if needed

Comment (rmcabezon) [reply sph/include/sph/hydro_ve/momentum_energy_kern.hpp:176]: > This change makes the [MomentumEnergy.JLoop](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L172) test to fail:
> 
> > EXPECT_NEAR(maxvsignal, 4.5535959472046174e7, 1e-6);
> 
> ```
> The difference between maxvsignal=26490876.319252387 and
> 4.5535959472046174e7 is 19045083.152793787, which exceeds 1e-6
> ```
> 
> while the previous code passes the test:
> 
> ```
>         T vijsignal = ci + cj - T(3) * wij;
>         maxvsignal= 45536000
> ```
> 
> A fix would be to modify [L172](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L172) and [L194](https://github.com/unibas-dmi-hpc/SPH-EXA/blob/develop/sph/test/hydro_ve/momentum_energy_kern.cpp#L194) of the test OR rectify [sph/include/sph/hydro_ve/momentum_energy_kern.hpp](https://github.com/unibas-dmi-hpc/SPH-EXA/pull/387/files/8f891e63f19d67793e6b8701624fa4bef88fc2d9#diff-18c0a64c44c98b69666874dd53ae33110f32a0292bacbc46b85ff8a05b693269)

This is expected, as vijsignal is precisely where the modification was done. I expected a bit more than a factor 2 reduction, which is what we get.

Comment (jgphpc) [reply sph/include/sph/hydro_ve/momentum_energy_kern.hpp:176]: https://github.com/unibas-dmi-hpc/SPH-EXA/pull/387/commits/9958a657409f0654ad29cac808f7862a30b48490

**Review by jgphpc — APPROVED** [review]

**Pull request merged**